dapp/src/components/buySell/BuySellWidget.js

```diff
@@ -339,15 +340,46 @@ const BuySellWidget = ({
 
       setBuyWidgetState(`${prependStage}waiting-network`)
       onResetStableCoins()
-      storeTransaction(result, `mint`, mintedCoins.join(','), {
-        usdt,
-        dai,
-        usdc,
-        ousd: totalOUSD,
-      })
+      const receipt = await rpcProvider.waitForTransaction(result.hash)
+      const data = {
+        usdt: 0,
+        dai: 0,
+        usdc: 0,
+        ousd: 0,
+      }
+      await Promise.all(
+        receipt.logs.map(async (log) => {
+          if (
+            log.address.toLowerCase() === addresses.mainnet.USDT.toLowerCase()
+          ) {
+            const value = ethers.BigNumber.from(log.data)
+            const decimals = await usdtContract.decimals()
```

Comment:
These are constants and will never change. We can probably save a contract call here

(applies to a few more cases below)